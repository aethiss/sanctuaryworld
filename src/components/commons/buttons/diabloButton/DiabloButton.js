import React, { useEffect } from 'react'
import PropTypes from 'prop-types'

// Material-Ui && Styles
import { makeStyles } from '@material-ui/core/styles'
import { OldFenris } from '../../commonStyles/fontFamily'

// Helpers
import { cacheImage } from '../../../../libs/cacheImages'

// Next
import Link from 'next/link'

const initStyles = ({ isDisabled, enableOnOver }) => {
  const baseBackground = () => {
    return isDisabled ?
      {
        background: 'url("/images/buttons/diablo-buttons-disable.png")',
        backgroundSize: 'contain',
        backgroundRepeat: 'no-repeat',
      } :
      {
        background: 'url("/images/buttons/diablo-buttons-base.png")',
        backgroundSize: 'contain',
        backgroundRepeat: 'no-repeat',
      }
  }

  const activeOver = (src) => {
    if (enableOnOver || !isDisabled) {
      return {
        background: `url(${src})`,
        backgroundSize: 'contain',
        backgroundRepeat: 'no-repeat',
      }
    }
    return {}
  }

  const styles = {
    base: {
      ...baseBackground(),
      ...OldFenris,
      width: '230px',
      height: '87px',
      padding: '0',
      color: '#FFF',
      border: 'none',
      outline: 'none',
      '&:hover': activeOver('/images/buttons/diablo-buttons-hover.png'),
      '&:active': activeOver('/images/buttons/diablo-buttons-press.png'),
    },
  }

  return makeStyles(styles)
}

const DiabloButton = ({ text, isDisabled, enableOnOver, anchorLink, action }) => {
  const classes = initStyles({ isDisabled, enableOnOver })()

  useEffect(() => {
  const imgs = [
    '/images/buttons/diablo-buttons-disable.png',
    '/images/buttons/diablo-buttons-hover.png',
    '/images/buttons/diablo-buttons-press.png'
  ]
    cacheImage(imgs)
  }, [])

  const renderButton = () => (
    <button
      className={classes.base}
      onClick={action}
    >
      {text}
    </button>
  )

  return (
    <>
      { anchorLink ? (
        <Link href={anchorLink}>
          <a>
            {renderButton()}
          </a>
        </Link>
      ) : renderButton() }
    </>
  )
}

DiabloButton.propTypes = {
  text: PropTypes.string,
  isDisabled: PropTypes.bool,
  enableOnOver: PropTypes.bool,
  anchorLink: PropTypes.string,
  action: PropTypes.func,
}

DiabloButton.defaultProps = {
  text: '',
  action: () => {},
}

export default DiabloButton
