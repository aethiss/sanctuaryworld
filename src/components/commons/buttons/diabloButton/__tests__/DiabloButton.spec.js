import React from 'react'
import { createMount } from '@material-ui/core/test-utils'
import { configure } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

configure({ adapter: new Adapter() })

import DiabloButton from '../DiabloButton'

describe('DiabloButton Component', () => {
  describe('Render', () => {
    it('[DiabloButton] Default render', () => {
      const testRenderer = createMount()(<DiabloButton />)
      expect(testRenderer).toMatchSnapshot()
      // const testInstance = testRenderer.root
      // const button = testInstance.findByType('button')
      // expect(button?.props?.style).toMatchObject({
      //   background: 'url("images/buttons/diablo-buttons-base.png")'
      // })
    })
    it.skip('[DiabloButton] Render disabled button', () => {
      const testRenderer = createMount()(<DiabloButton isDisabled={true} />)
      const testInstance = testRenderer.root
      const button = testInstance.findByType('button')
      expect(button?.props?.style).toMatchObject({
        background: 'url("/images/buttons/diablo-button-disable.png") no-repeat'
      })
    })
  })
})
