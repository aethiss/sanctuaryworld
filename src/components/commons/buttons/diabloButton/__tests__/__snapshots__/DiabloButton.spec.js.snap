// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiabloButton Component Render [DiabloButton] Default render 1`] = `ReactWrapper {}`;
